main.c

```diff
@@ -417,15 +417,20 @@ has_prefix (const char *str, const char *pref)
 static int
 set_fd_opaque (int fd)
 {
-  if (fsetxattr (fd, PRIVILEGED_OPAQUE_XATTR, "y", 1, 0) < 0)
+  cleanup_close int opq_whiteout_fd = -1;
+  int ret;
+
+  ret = fsetxattr (fd, PRIVILEGED_OPAQUE_XATTR, "y", 1, 0);
+  if (ret < 0)
     {
       if (errno == ENOTSUP)
-        return 0;
+        goto create_opq_whiteout;
       if (errno != EPERM || fsetxattr (fd, OPAQUE_XATTR, "y", 1, 0) < 0 && errno != ENOTSUP)
           return -1;
     }
-
-  return 0;
+ create_opq_whiteout:
```

Comment:
It looks like you are creating the opq_whiteout even if the fsetxattr above is successful?  Is this intentional

Reply:
yes, I am not sure we still want the `OPAQUE_XATTR` which is fuse-overlayfs specific.  I've not removed it yet as it might break existing images.